### PR TITLE
feat: Create sign-in page.

### PR DESCRIPTION
**Pull request by MjxOro**
This page was created by re-using the Header, Greeting, and Inputbox components. It is similar to the sign-up page but with fewer input boxes.

Instructions to test page:
- On App.js, import the AuthenticationPage component.
- AuthenticationPage takes a prop "isSignin" which is a boolean type. 
- replace HomePage component with AuthenticationPage component.

Extras:
- Greeting component takes props: "title", "linkMessage", "link" which are all string types.
- GreenButton component takes prop: "message" which is a string type.
- InputBox component takes prop: "placeholder" which is a string type.
- AuthenticationForm component takes "placeholders" which is a string[] (string array) type and "buttonMessage" which is a string type.

Screen-shot of the page below:
Screen Shot 2022-07-12 at 7 47 07 PM

### CODE REVIEW

**Comment by YoussefLaunchUp** [comment]
Can you change the name of the PR to follow the convention?

File: src/components/Greeting.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+function Greeting({title,link}){
+  return(
+    <div className="flex flex-col justify-center items-center">
+      <h1 className="text-3xl my-4 md:text-6xl lg:text-12xl">{title}</h1>
+      <a href="#" className="text-main-green">{link}</a>
```
Comment (YoussefLaunchUp) [inline src/components/Greeting.js]: Where the link points to should also be a param. (even if whats being passes in right now is just "#")

File: src/components/SignInPage/Form.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import InputBox from '../InputBox';
+
+function Form(){
+  return(
+    <form className="w-[30%] flex flex-col justify-center items-center my-4" >
+      <InputBox placeholder={"Email"} />
+      <InputBox placeholder={"Password"} />
+      <button type="button" className="self-end focus:outline-none text-white bg-main-green hover:bg-green-800 focus:ring-4 focus:ring-green-300 font-medium rounded-md text-sm px-5 py-2.5">Sign in</button>
```
Comment (YoussefLaunchUp) [inline src/components/SignInPage/Form.js]: This seems to be mostly the same button as the sign up page. Lets make it into a component.

File: src/components/SignInPage/Form.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import InputBox from '../InputBox';
+
+function Form(){
```
Comment (YoussefLaunchUp) [inline src/components/SignInPage/Form.js]: This Form component is similar to the Sign Up page Form. Lets merge them into one component that takes in params for what input boxes are needed. Lets call it authentication form.

File: src/components/SignUpPage/SignUpPage.js [hunk 4]
```diff
@@ -7,7 +7,7 @@ function SignUpPage(){
     <div>
```
Comment (YoussefLaunchUp) [inline src/components/SignUpPage/SignUpPage.js]: Given the similarities to the SignIn page, I recommend merging the two allowing for more reusability and less repetition of code.

**Review by YoussefLaunchUp — CHANGES_REQUESTED** [review]

**Comment by YoussefLaunchUp** [comment]
Also Add instructions as to how to test the code to ensure it is working if something needs to be changed.

File: src/components/SignInPage/Form.js (hunk 3 above)
Comment (MjxOro) [reply src/components/SignInPage/Form.js]: I created the auth form component which takes an array of strings for the input box placeholder and another parameter that takes the message for the button.

File: src/components/Greeting.js (hunk 1 above)
Comment (MjxOro) [reply src/components/Greeting.js]: Okay, I added the param to take the message and link.

File: src/components/SignInPage/Form.js (hunk 2 above)
Comment (MjxOro) [reply src/components/SignInPage/Form.js]: I created a green button component that takes a string for the message on the button.

File: src/components/SignUpPage/SignUpPage.js (hunk 4 above)
Comment (MjxOro) [reply src/components/SignUpPage/SignUpPage.js]: I created an AuthenticationPage component which takes a param if ever the user chooses sign-up or sign-in, the component will fill the page accordingly.

**Review by YoussefLaunchUp — APPROVED** [review]
Great work on the PR description!

**Pull request merged**